pkg/external/pod.go

```diff
@@ -109,51 +91,28 @@ func createFencingJob(action string, machine *clusterv1.Machine, fencingConfig *
 	}, nil
 }
 
-func volumeNameMap(r rune) rune {
-	switch {
-	case r >= 'A' && r <= 'Z':
-		return 'a' + (r - 'A')
-	case r >= 'a' && r <= 'z':
-		return r
-	case r >= '0' && r <= '9':
-		return r
-	default:
-		return '-'
-	}
-}
-
-func processSecrets(fencingConfig *v1alpha1.FencingConfig, c *v1.Container) []v1.Volume {
-	volumes := []v1.Volume{}
-	for key, s := range fencingConfig.Secrets {
-
-		// volumeName must contain only a-z, 0-9, and -
-		volumeName := strings.Map(volumeNameMap, fmt.Sprintf("secret-%s", key))
-		mount := fmt.Sprintf("%s/%s-%s", secretsDir, s, key)
-		data := fmt.Sprintf("%s/%s", mount, key)
-
-		// Create volumes for any sensitive parameters that are stored as k8s secrets
-		volumes = append(volumes, v1.Volume{
+func processSecret(fencingConfig *v1alpha1.FencingConfig, c *v1.Container) []v1.Volume {
+	// Create volumes for any sensitive parameters that are stored as k8s secrets
+	volumeName := "fencing-secret"
```

Comment:
Doesn't this assume a single secret?
Not sure thats a safe assumption

Reply:
> How would an ansible (or bespoke customer) version of this look?
> Would you suggest their container inherit from `fence-provision-manager`, or implement their own argument handling?

The end user must get verified and tested container that will be the entrypoint for all fence provision operation, he still will have possibility to create his own container that will include fencing and provisioning logic, but it will be on his own risk.